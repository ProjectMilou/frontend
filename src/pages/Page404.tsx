import React from 'react';
import { Link, RouteComponentProps } from '@reach/router';
import {
  Container,
  Box,
  Typography,
  Button,
  makeStyles,
  Theme,
  createStyles,
} from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    button: {
      margin: theme.spacing(1),
      height: theme.spacing(6),
    },
  })
);

const Page404: React.FC<RouteComponentProps> = () => {
  const { button } = useStyles();
  const { t } = useTranslation();
  return (
    <Container maxWidth="md">
      <Box my={10}>
        <Typography variant="h1" align="center" color="primary">
          404
        </Typography>
        <Typography variant="h6" align="center">
          {t('error.invalidPage')}
        </Typography>
        <Box mt={2} style={{ display: 'flex', justifyContent: 'center' }}>
          <Button
            className={button}
            variant="contained"
            color="primary"
            component={Link}
            to="/"
          >
            {t('error.action.home')}
          </Button>
        </Box>
      </Box>
    </Container>
  );
};

export default Page404;
