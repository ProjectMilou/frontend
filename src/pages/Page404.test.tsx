import { fireEvent, render, screen } from '@testing-library/react';
import * as React from 'react';
import Page404 from './Page404';

describe('404 Page', () => {
  test('should render 404-page', async () => {
    const { container } = render(<Page404 />);
    expect(container).toMatchSnapshot();
  });

  test('should navigate to home', async () => {
    render(<Page404 />);
    const goToHome = screen.getByText(/error.action.home/i);
    fireEvent.click(goToHome);
    expect(window.location.pathname).toBe('/');
  });
});
