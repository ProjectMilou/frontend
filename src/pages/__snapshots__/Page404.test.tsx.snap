// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page should render 404-page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-colorPrimary MuiTypography-alignCenter"
      >
        404
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
      >
        error.invalidPage
      </h6>
      <div
        class="MuiBox-root MuiBox-root-3"
        style="display: flex; justify-content: center;"
      >
        <a
          aria-current="page"
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
          href="/"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            error.action.home
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
